mozilla/bedrock
# Track VPN button clicks as GA4 "begin_checkout" event

## One-line summary

Track VPN button clicks in GA4 as "begin_checkout" event

## Significant changes and points to review

- add data-attributes to VPN buttons with item information in JSON format
- add JS functions which consume data-attributes to get item information
- listen for button clicks on VPN landing page
- tests, tests, tests across the board

## Issue / Bugzilla link

n/a

## Testing

Demo server URL: (or None)

https://www-demo3.allizom.org/en-US/products/vpn/
https://www-demo3.allizom.org/en-US/products/vpn/pricing/

To test this work:

Bedrock code:
- [x] existing VPN tests are updated and passing
- [x] new tests provide adequate coverage
- [x] good checks exist so things don't break if the data is missing or malformed
- [x] `data-ga-item` attributes are added to VPN buttons
- [x] `handleLinkWithItemData` is fired when the "Get X plan" buttons are clicked on VPN landing page
- [x] `handleLinkWithItemData` is fired when the "Get X plan" buttons are clicked on VPN pricing page

GTM:
- Switch to the workspace in the www.mozilla.org workspace called "VPN begin_checkout".
- Click "preview" to open the tag assistant website.
- Add the domain "https://www-demo3.allizom.org/" and click it
- Disable add-ons and toggle off DNT, etc until tag assistant indicates it has a connection
- Navigate to the vpn page and click a "Get X plan" button
- Return to the tag assistant window and click the 'begin_checkout' event
- [x] Verify Data Layer contains the event and item array

GA4:
- Switch to the www.mozilla.org - GA4 Testing property
- Click the Admin menu item (bottom left)
- Click the DebugView link (last item in the second column before the colourful product links)
- Your tag assistant session should already have sent data here, you might have to pick a device out of the drop down until "Debug Device"
- [x] Verify the begin_checkout event has been logged
- [x] Verify the event has a currency and value
- [x] Verify the event has an item associated with it

Misc:
- [x] currency is accurate for USD, EUR, and CHF transactions
- [x] item category is vpn
- [x] item name is 'vpn' unless buying the bundle, then it is 'vpn + relay' (I'm open to alternate suggestions for the un-bundled purchase. Basic? Default? Base? Standard? For Relay these will be limited, email, email + phone)
- [x] variant is monthly or yearly as appropriate
- [ ] price and discount are correct (until they reach GA4 - the item price is doing something weird in GA4 I don't know why, because the begin_checkout value is fine. I was hoping to push this and see if it's a problem in the debug UI and not the data)



Tests:
